_posts/en/posts/2016-06-06-compact-blocks-faq.md

```diff
@@ -0,0 +1,144 @@
+
+Furthermore, the experiments conducted and data collected using the first version of compact blocks will inform the design of the future improvements we expect to be more competitive with the FRN.
+
+## Who benefits from compact blocks?
+
+* Full node users who want to relay transactions but who have limited internet bandwidth. If you simply want to save the most bandwidth possible while still relaying blocks to peers, there is a `blocksonly` mode already available starting in Bitcoin Core v0.12.  Blocks-only mode only receives transactions when they're included in a block, so there is no extra transaction overhead.
+
+* The network as a whole. Decreasing block propagation times on the p2p network creates a healthier network with a better baseline relay security margin. 
+
+## What is the timeline on coding, testing, reviewing and deploying compact block propagation?
+
+The first version of compact blocks has been assigned BIP152, has a working implementation, and is being actively tested by the developer community. 
+
+- BIP152: <https://github.com/bitcoin/bips/blob/master/bip-0152.mediawiki>
+- Reference implementation: <https://github.com/TheBlueMatt/bitcoin/tree/udp>
+
+## How can this be adapted for miners?
+
+In order to get miner adoption additional improvements to the compact block scheme must be made. These are two-fold:
+
+- First, replace TCP transmission of block information with UDP transmission.
+
+- Second, handle dropped packets and pre-emptively send missing transaction data by using forward error correction (FEC) codes. 
+
+UDP transmission allows data to be sent by the server and digested by the client as fast as the path allows, without worrying about intermittent dropped packets. A client would rather receive packets out of order to construct the block as fast as possible but TCP does not allow this. 
+
+In order to deal with the dropped packets and receiving non-redundant block data from multiple servers, FEC codes will be employed. A FEC code is a method of transforming the original data into a redundant code, allowing lossless transmission as long as a certain percentage of packets arrive at its destination, where the required data is only slightly larger than the original size of the data.
+
+This would allow a node to begin sending a block as soon as it receives them, and allow the recipients to reconstruct blocks being streamed from multiple peers simultaneously. All of this work continues to build on the compact blocks work already completed. This is a medium-term extension, and development can be watched here: 
+
+- Reference implementation: <https://github.com/TheBlueMatt/bitcoin/tree/udp-wip>
```

Comment:
This is targeted at RelayNetwork-NG now...less a useful example of miners switching to something that could be run on a p2p network more a system for building private relay networks, it has a mode to work among untrusted nodes, but much less efficiently,